docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+    users: list[list] = [
+        ["Danilo Sousa", "danilo@example.com", "Male"],
+        ["Zahra Ambessa", "zahra@example.com", "Female"],
+    ]
+```
+
+To iterate over a state var that is a list, we use the [`rx.foreach`]({docs.components.rendering_iterables.path}) function to render a list of components. The `rx.foreach` component takes an `iterable` (list, tuple or dict) and a `function` that renders each item in the `iterable`.
+
+Here the render function is `show_user` which takes in a single user and returns a `table.row` component that displays the users name, email and gender.
+
+```python exec
+class State1(rx.State):
+    users: list[list] = [
+        ["Danilo Sousa", "danilo@example.com", "Male"],
+        ["Zahra Ambessa", "zahra@example.com", "Female"],
+    ]
+
+def show_user1(person: list):
+    """Show a person in a table row."""
+    return rx.table.row(
+        rx.table.cell(person[0]),
+        rx.table.cell(person[1]),
+        rx.table.cell(person[2]),
+    )
+```
+
+```python eval
+rx.table.root(
+        rx.table.header(
+            rx.table.row(
+                rx.table.column_header_cell("Name"),
+                rx.table.column_header_cell("Email"),
+                rx.table.column_header_cell("Gender"),
+            ),
+        ),
+        rx.table.body(
+            rx.foreach(
+                State1.users, show_user1
+            ),
+        ),
+        variant="surface",
+        size="3",
+)
+```
+
+
+```python
+class State(rx.State):
+    users: list[list] = [
+        ["Danilo Sousa", "danilo@example.com", "Male"],
+        ["Zahra Ambessa", "zahra@example.com", "Female"],
+    ]
+
+def show_user(person: list):
+    """Show a person in a table row."""
+    return rx.table.row(
+        rx.table.cell(person[0]),
+        rx.table.cell(person[1]),
+        rx.table.cell(person[2]),
+    )
+
+def index() -> rx.Component:
+    return rx.table.root(
+        rx.table.header(
+            rx.table.row(
+                rx.table.column_header_cell("Name"),
+                rx.table.column_header_cell("Email"),
+                rx.table.column_header_cell("Gender"),
+            ),
+        ),
+        rx.table.body(
+            rx.foreach(
+                State.users, show_user
+            ),
+        ),
+        variant="surface",
+        size="3",
+)
+```
+
+As you can see the output above looks the same as before, except now the data is no longer static and can change with user input to the app.
+
+### Using a proper class structure for our data
+
+So far our data has been defined in a list of lists, where the data is accessed by index i.e. `user[0]`, `user[1]`. This is not very maintainable as our app gets bigger. 
+
+A better way to structure our data in Reflex is to use a class to represent a user. This way we can access the data using attributes i.e. `user.name`, `user.email`.
+
+In Reflex when we create these classes to showcase our data, the class must inherit from `rx.Base`.
```

Comment:
Why should I inherit from rx.Base? What usage of this class would necessitate that. Because there doesn't seem anything special about it right now

Reply:
added more information here (think we should add a follow up with a link to a doc with more rx.base)

Reply:
Hmm maybe switch the ``show_user'' paragraph with the one after it. Or just merge this new paragraph with the one that introduces rx.base, for concision.